shared/json_convertible.rb

```diff
@@ -29,7 +29,7 @@ def to_object_dictionary(ignore_instance_variables: [])
           if self.instance_variable_get(var).kind_of?(Array)
             object_array = []
             self.instance_variable_get(var).each do |obj|
-              object_array << obj.to_object_dictionary
+              object_array << (obj.kind_of?(Hash) ? obj : obj.to_object_dictionary)
```

Comment:
Is this change still necessary?

Reply:
Ahh, yeah - the change is still necessary because I'm serializing the `ProviderCredential`s to a `Hash` first, and then serializing the `User`. I documented the changes in my [most recent commit](https://github.com/fastlane/ci/pull/221/commits/3c3d7e3ba9ad9c6e0c894658c35342e0d5c6c02f). When you get a chance, could you please let me know if the documentation is good, or if there is anything I should change with the documentation or the code? Thank you!

Reply:
I'm kind of confused, because it can happen that many types does not include that method. So I think the smartest way to proceed is checking if the `obj.class.include?(JSONConvertible)`. What do you think?

Reply:
Ahh, I like that suggestion much better than what I implemented. Good call!

Reply:
Is like I forgot the safe fallback to types that does not include `JSONConvertible`. Could you add a test case that covers this case? 😄

Reply:
Will do! :)